.github/workflows/automated-release.yaml

```diff
@@ -0,0 +1,52 @@
+name: Automated Release
+
+permissions:
+  id-token: write
+  contents: write
+
+on:
+  push:
+    branches:
+      - release
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: pnpm/action-setup@v4
+        with:
+          version: 9.15.0
+      - uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+          registry-url: https://registry.npmjs.org/
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build the project
+        run: pnpm run build
+
+      - name: Publish to npm
+        run: pnpm publish -r --access public --publish-branch release
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NPM_CONFIG_PROVENANCE: true
+
+      - name: Get version from package.json
+        id: get_version
+        run: |
+          VERSION=$(node -p "require('./package.json').version")
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Create Git Tag
```

Comment:
Would `npx bumpp` also do the job for this?

Reply:
Ah, I see `bumpp` is part of the release script which is part of the build script.

So there should already be the new tag on this point. Or am I wrong?

`bumpp` creates the tags and usually pushes the tags too, but i don't know if the pushing works in a workfkow.

Reply:
Yes, `bumpp` supports tag and push, and then needs to monitor tag to publish. What I want is to monitor release branch to build, publish, and then tag. So I will remove `bumpp`.

Reply:
Okay, that sounds good :+1: 🙂